File: ios/RNGiphyMediaViewManager.m
#import "React/RCTViewManager.h"

@interface RCT_EXTERN_REMAP_MODULE(GiphyReactNativeMediaView, RNGiphyMediaViewManager, RCTViewManager)
  RCT_EXPORT_VIEW_PROPERTY(autoPlay, BOOL)

  RCT_EXPORT_VIEW_PROPERTY(media, NSDictionary)

  RCT_EXPORT_VIEW_PROPERTY(renditionType, NSString)

  RCT_EXPORT_VIEW_PROPERTY(resizeMode, NSString)

  RCT_EXTERN_METHOD(pause:(nonnull NSNumber *)node)

  RCT_EXTERN_METHOD(resume:(nonnull NSNumber *)node)
@end

File: ios/RNGiphyMediaViewManager.swift
@objc(RNGiphyMediaViewManager)
class RNGiphyMediaViewManager: RCTViewManager {
  override static func requiresMainQueueSetup() -> Bool {
    return true
  }

  override func view() -> UIView! {
    return RNGiphyMediaView()
  }

  func findView(_ node: NSNumber, completion: @escaping (_ result: RNGiphyMediaView?) -> Void) {
    DispatchQueue.main.async { [weak self] in
      guard let self = self else {
        return
      }

      let view = self.bridge.uiManager.view(forReactTag: node) as! RNGiphyMediaView?
      completion(view)
    }
  }

  @objc func pause(_ node: NSNumber) {
    findView(node) { view in
      view?.pause()
    }
  }

  @objc func resume(_ node: NSNumber) {
    findView(node) { view in
      view?.resume()
    }
  }
}
